homeassistant/components/sensor/zha.py

```diff
@@ -198,6 +198,11 @@ def unit_of_measurement(self):
         """Return the unit of measurement of this entity."""
         return 'W'
 
+    @property
+    def force_update(self) -> bool:
+        """Force update this entity."""
+        return True
```

Comment:
Should `MeteringSensor` also be "force updated"?

Reply:
No, it only reports in if it changes

Reply:
Can you ELI5 when "force_update" should be used and when it shouldn't? I was reading #2365 and per https://github.com/home-assistant/home-assistant/pull/2365#issuecomment-228544703 I thought it was meant more for devices with `should_poll = False` 

I'm still trying to comprehend the difference in behavior from the `MeteringSensor`. `MeteringSensor` is pushing updates to HA and when it receives attribute_report, it means the value changed and so it should be updated, correct?

`ElectricalMeasurementSensor` is polled, so you do read it every 30s, but if it has not changed since last reading, why does it need to be force updated? is it just to indicate that it was the most recent reading via "last updated" attribute?

I'm just thinking if we should do "force update" for binary_sensor.zha on/off sensors, since like with ZWave, those also can send "On" command even if the current status is "On"

Reply:
You're right that if an entity is `should_poll = False`, it is most likely wanted to use `force_update = True`. Since the `MeteringSensor` will only report to Home Assistant when the state changes, there is no need to use `force_update`, as the state will always have changed.